### ISSUE TITLE
Autopilot Aircon Unit 

### ISSUE THREAD

**Issue by tehfonz** [dgaust/smarterzones#6]
Would love to build in an enhancement to turn the aircon unit itself on/off depending on the following:
If all targeted zones are within their target temperature range - Turn AirCon Off _unless_ AirCon has been ON Less Than _x_ minutes (using Number Input Helper).
If any of the targeted zones go outside of target temp range - Turn AirCon ON _unless_ AirCon has been OFF Less Than _y_ minutes (using Number Input Helper).
I should have added, all of this is triggered by or controlled by a Toggle to tell the app to control the AirCon unit as well as zones.
Basically want to avoid the situation where the aircon turns on / off rapidly, killing motor or something else.
Hope this is doable.
Cheers,
_Fonz

**Comment by dgaust** [comment]
No guarantees, but I can look into it. I've specifically not included that functionality as I always wanted manual control as the baseline, but can look at how it might work.

**Comment by tehfonz** [comment]
Totally understand, might constitute a complete rewrite.

**Comment by dgaust** [comment]
So I've started some work on this, at this stage it will only turn the air conditioner on and set it to heat/cool mode when a temperature sensor passes a particular threshold (as the balancing aspect of the controller should manage the in zone temp aspect).

Bascially you set a temp sensor that you want to trigger the action, then a temp you want it to trigger on (both upper and lower) in the apps.yaml

trigger_temp_sensor: sensor.living_room_temperature
trigger_temp_upper: 27
trigger_temp_lower: 18

Once the temp of this sensor exceeds (or drops below) the triggers it will check to see if the airconditioner is off, and if it is it will turn it on.

It's fairly dumb at the moment, in that it will happen even if no-one is home (I will probably add presence detection or some other method of sanity), but you can try it using the code from this branch https://github.com/dgaust/smarterzones/tree/auto_control

**Comment by tehfonz** [comment]
Thanks so much for adding this. I will add the branch to my HA, however we don't really use the air-con much during the colder months, and if we do it's all manual with just a quick 10-20 min run. 
Really appreciate the effort!

**Comment by dgaust** [comment]
There were a few issues with the previous version, I have made changes to the code and tested a little more extensively. A couple of additional settings are required in the apps.yaml section, but that has also been updated in this branch.
